### PR TITLE
Check __cplusplus is defined before checking its value

### PR DESCRIPTION
**Pull request by davidjmccann**
If __cplusplus isn't defined when omrcomp.h is included, it can cause a compiler warning to be emitted:

include_core/omrcomp.h:612:6: error: "__cplusplus" is not defined, evaluates to 0 [-Werror=undef]
  612 | #if (__cplusplus >= 201103L)

### CODE REVIEW

File: include_core/omrcomp.h
Comment (babsingh) [inline include_core/omrcomp.h:612]: The corresponding `#endif` also needs to be updated:
```
#endif /* defined(__cplusplus) && (__cplusplus >= 201103L) */
```

**Review by babsingh — CHANGES_REQUESTED** [review]
Can you please provide more details on the compiler and platform where this error was seen?

There are three other locations where this error can occur:
```
compiler/codegen/CCData_inlines.hpp:#if (__cplusplus >= 201103L)
compiler/codegen/CCData_inlines.hpp:#if (__cplusplus >= 201103L)
compiler/codegen/CCData_inlines.hpp:#if (__cplusplus >= 201103L)
```

For completeness, these locations should also be updated.

**Comment by babsingh** [comment]
The `eclipsefdn/eca` check has also failed. To pass this check, you need to create an account at Eclipse and sign the ECA. Also, the email address in the commit message should match the email address used with your Eclipse account. Instructions: https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md#legal-considerations.

File: include_core/omrcomp.h
Comment (davidjmccann) [reply include_core/omrcomp.h:612]: Thank you, you are quite right. I believe I've now amended these other references as requested, and added the signature block to the commit message.

**Review by babsingh — APPROVED** [review]
LGTM. Approving.

This question was missed: `Can you please provide more details on the compiler and platform where this error was seen?` Currently, this error is not seen in the PR builds. The answer will document the conditions to reproduce the error.

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
[zOS failures](https://ci.eclipse.org/omr/job/PullRequest-zos_390-64/3425/) are unrelated to this PR and due to https://github.com/eclipse/omr/issues/6368.

**Comment by jdekonin** [comment]
@babsingh the new z/OS nodes were enabled for some reason.  I believe the failures in testing are expected and due to path issues that PR #6250 resolves.

**Comment by davidjmccann** [comment]
@babsingh I was using Fedora 35, which by default uses gcc 11. I pulled omrcomp.h into a C program, rather than a CPP one, and compiling caused that because building with gcc rather than g++ will not set __cplusplus.

**Comment by davidjmccann** [comment]
Excuse my ignorance, but what needs to happen next?

**Comment by babsingh** [comment]
> Excuse my ignorance, but what needs to happen next?

One of the committers (e.g. @0xdaryl) should review and merge the PR soon.

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Congratulations on your first contribution!

**Pull request merged**
